presto-main/src/main/java/com/facebook/presto/execution/UseTask.java

```diff
@@ -35,6 +41,8 @@
 public class UseTask
         implements SessionTransactionControlTask<Use>
 {
+    String catalog;
+    String schema;
```

Comment:
nit: please add a new line after this

Reply:
Instead of adding these variables to the class, can you just call `hasCatalogAccess` from `checkAndSetCatalog` before the catalog is set? Also same for `hasSchemaAccess` in  `checkAndSetSchema` before  `stateMachine.setSetSchema(schema)`